data/campaigns/The_South_Guard/scenarios/02_Proven_by_the_Sword.cfg

```diff
@@ -549,24 +555,43 @@
         [/kill]
 
         [if]
-            [have_unit]
-                id=Minister Hylas
-                x,y=$x2,$y2
-            [/have_unit]
-
+            [not]
+                [have_unit]
+                    id="Minister Hylas"
+                [/have_unit]
+            [/not]
             [then]
-                [message]
-                    speaker=Deoran
-                    message= _ "How can a bandit have summoned undead?"
-                [/message]
+                {MINISTER_HYLAS}
             [/then]
+        [/if]
 
-            [else]
-                [message]
-                    speaker=second_unit
-                    message= _ "How can a bandit have summoned undead?"
-                [/message]
-            [/else]
+        [if]
+            [variable]
+                name=citadel_secured
+                equals="yes"
+            [/variable]
+            [then]
+                [if]
+                    [have_unit]
+                        id=Minister Hylas
+                        x,y=$x2,$y2
+                    [/have_unit]
+
+                    [then]
+                        [message]
+                            speaker=Deoran
+                            message= _ "How can a bandit have summoned undead?"
+                        [/message]
+                    [/then]
+
+                    [else]
+                        [message]
+                            speaker=second_unit
+                            message= _ "How can a bandit have summoned undead?"
```

Comment:
Is it possible for this unit to be female? (ie, is there any recruitable unit that can be female?)